SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialLoadBalancingHandler.cpp

```diff
@@ -46,6 +47,34 @@ FSpatialLoadBalancingHandler::EvaluateActorResult FSpatialLoadBalancingHandler::
 	if (NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID))
 	{
 		AActor* NetOwner = GetReplicatedHierarchyRoot(Actor);
+
+		if (AController* Controller = Cast<AController>(Actor))
+		{
+			TArray<UActorComponent*> Components;
+			Controller->GetComponents(URemotePossessionComponent::StaticClass(), Components);
+			if (Components.Num() == 1)
+			{
+				if (URemotePossessionComponent* Component = Cast<URemotePossessionComponent>(Components[0]))
+				{
+					if (NetDriver->LockingPolicy->IsLocked(Actor))
+					{
+						UE_LOG(LogRemotePossessionComponent, Warning, TEXT("Actor %s (%llu) cannot migrate because it is locked"),
+							   *Actor->GetName(), EntityId);
+						Component->DestroyComponent();
+						return EvaluateActorResult::None;
+					}
+
+					VirtualWorkerId TargetVirtualWorkerId;
+					if (EvaluateRemoteMigrationComponent(NetOwner, Component->Target, TargetVirtualWorkerId))
```

Comment:
Worth making Target an accessor function that a user could override? Useful if users wanted to extend this functionality.

Reply:
What kind of scenario do you have in mind? This is opening the door to user mistakes if they somehow source their target from some place they forgot to/cannot replicate.

Reply:
Not quite sure understand. I guess your concern is Target "why not have a `GetTargetPawn()` in RemotePossessionComponent?"
Yes, I think about it before.
In my mind, I want to make my implementation of RemotePossessionComponent in BluePrint or C++ if I was the end-user.
I want to set the Target anywhere and visit it anywhere.
We can have a getter and a setter for Target, and make them blueprint callable. After doing that, the code will more complicate than only have a Target.

Reply:
Okay - can keep as is.